### PR TITLE
add api to handle input conditon and mask

### LINKED ISSUE #7720: Add more flexible search option with active requests
provide api  when user wants to search specific conditions over active request. 

i.e /wmstatsserver/data/filterted_requests?RequestStatus=running-closed&RequestStatus=running-open&RequestType=TaskChain&SubRequestType=RelVal&mask=RequestName&mask=PrepID

This will return only RequestName and PrepID for  TaskChain, RelVal  request on running-open and running-closed status.

### PR DESCRIPTION
**Pull request by ticoann**
fixes #7720

### CODE REVIEW

**Comment by ticoann** [comment]
Also possible to search task/stepchain level parameters. (without mask it returns every parameters)

```
https://reqmgr2-dev.cern.ch/wmstatsserver/data/filtered_requests?RequestStatus=completed&RequestStatus=assigned&PrepID=TEST-Step3-RunIISummer15wmLHEGS-00744&mask=ConfigCacheID&mask=PrepID

returns

{"result": [
 {
  "PrepID": [
    "TEST-Step3-RunIISummer15wmLHEGS-00744", 
    "TEST-Step1-RunIISummer15wmLHEGS-00744", 
    "TEST-Step2-RunIISummer15wmLHEGS-00744"
  ], 
  "RequestName": "sryu_StepChain_MC_reqmgr2_170307_181320_6462", 
  "ConfigCacheID": [
    "f7e311f2c6b5a0884faea133990edcbf", 
    "526ca0745cd309b7cfef9f23b3d43acb", 
    "f7e311f2c6b5a0884faea133990f66cc"
  ]
}]}
```

**Comment by ticoann** [comment]
test this please

File: src/python/WMCore/ReqMgr/DataStructs/Request.py [hunk 1]
```diff
@@ -188,8 +188,12 @@ def andFilterCheck(self, filterDict):
             if not isinstance(value, list):
                 value = [value]
 
-            if key in self.data:
-                if self.data[key] not in value:
+            reqValue = self.get(key)
```
Comment (amaltaro) [inline src/python/WMCore/ReqMgr/DataStructs/Request.py]: Isn't it `self.data.get(key)`?

**Review by amaltaro — COMMENTED** [review]
Besides that one question, patch looks good.

File: src/python/WMCore/ReqMgr/DataStructs/Request.py (hunk 1 above)
Comment (ticoann) [inline src/python/WMCore/ReqMgr/DataStructs/Request.py]: actually self.get(key) is correct. so it can get the Task/StepChain param correctly.

Comment (amaltaro) [inline src/python/WMCore/ReqMgr/DataStructs/Request.py]: but looking here
https://github.com/ticoann/WMCore/blob/76a2f26ec11e60cf06afe4a9aa275a7eb70dd26b/src/python/WMCore/ReqMgr/DataStructs/Request.py#L122
don't we assign the whole request dict to self.data?

Comment (ticoann) [inline src/python/WMCore/ReqMgr/DataStructs/Request.py]: Yes but if we use that we can handle the Task/StepChain correctly.  i.e. If PrepID is set in Task field we can match it. So I am using this instead.

https://github.com/ticoann/WMCore/blob/76a2f26ec11e60cf06afe4a9aa275a7eb70dd26b/src/python/WMCore/ReqMgr/DataStructs/Request.py#L159

**Comment by ticoann** [comment]
test this please

**Pull request merged**